src/main/java/org/junit/internal/ComparisonCriteria.java

```diff
@@ -56,27 +68,31 @@ public void arrayEquals(String message, Object expecteds, Object actuals)
                 }
             }
         }
-    }
 
-    private boolean isArray(Object expected) {
-        return expected != null && expected.getClass().isArray();
+        if (actualsLength != expectedsLength) {
+            Object expected = getArrayElementOrSentinel(expecteds, expectedsLength, prefixLength);
+            Object actual = getArrayElementOrSentinel(actuals, actualsLength, prefixLength);
+            try {
+                Assert.assertEquals(expected, actual);
```

Comment:
I will let @marcphilipp decide, but if the previous code used fail with a hard-coded message, then I think we should do the same here, to minimize the observable change on behavior.